Add error explanation for E0755 (#76439)
rust-lang/rust: compiler/rustc_error_codes/src/error_codes/E0755.md

The `ffi_pure` attribute was used on a non-foreign function.

Erroneous code example:

```compile_fail,E0755
#![feature(ffi_pure)]

#[ffi_pure] // error!
pub fn foo() {}
# fn main() {}
```

The `ffi_pure` attribute can only be used on foreign functions which do not have

> effects except for their return value. You can find more information about it in
> the [unstable Rust Book].

Reviewer: ```suggestion
side effects. You can find more information about it in the [unstable Rust Book].
```
Returning value is probably not counted as side effects.

Author: I quoted the book.

Reviewer: Yes, I took from the book too.

Author: Then perfect. :)

```
#![feature(ffi_pure)]

extern "C" {
    #[ffi_pure] // ok!
    pub fn strlen(s: *const i8) -> isize;

> }

Reviewer: I believe this also work right?
```rust
#[ffi_pure]
extern "C" pub fn foo() {}
```

Author: I think it's better to have a bigger contrast. :)

Reviewer: Oh, I was just suggesting to show both methods so the readers know.
